draft-ietf-mls-protocol.md

```diff
@@ -1948,23 +1944,25 @@ provides forward secrecy and post-compromise security with regard to the sender
 of the Commit.  An Update proposal can be regarded as a "lazy" version of this
 operation, where only the leaf changes and intermediate nodes are blanked out.
 
-The `path` field of a Commit message MUST be populated if the Commit covers at
-least one Update or Remove proposal, i.e., if the length of the `updates` or
-`removes` vectors is greater than zero.  The `path` field MUST also be populated
-if the Commit covers no proposals at all (i.e., if all three proposal vectors
-are empty).  The `path` field MAY be omitted if the Commit covers only Add
-proposals.  In pseudocode, the logic for whether the `path` field is required is
-as follows:
+Group members MUST verify that the `proposals` field is sorted such that Update
+proposals come first, then Remove proposals, then Add proposals. The `path`
+field of a Commit message MUST be populated if the Commit covers at least one
+Update or Remove proposal, or if the Commit covers no proposals at all.  The
+`path` field MAY be omitted if the Commit covers only Add proposals. In
+pseudocode, the logic for validating a commit is as follows:
 
 ~~~~~
-haveUpdates = len(commit.Updates) > 0
-haveRemoves = len(commit.Removes) > 0
-haveAdds = len(commit.Adds) > 0
+onlyAdds = true
```

Comment:
Yes it does, line 1962 does that check

Reply:
line 1967 now

Reply:
In a related vein, this PR should:
1. Extend the `capabilities` extension to have `supported_proposals`
2. Add a note to the proposal processing step below that future proposal handling is defined by the future proposals.

Reply:
Added a note that the processing should be updated by extensions that add new proposals.

I don't want to add a `supported_proposals` vector, because proposals can't be supported piecemeal. If you support an extension and that extension adds several proposals, you'd need to support them all